main.go

```diff
@@ -69,10 +71,16 @@ func main() {
 
 	log.Infof("Dozzle version %s", version)
 	dockerClient := docker.NewClientWithFilters(args.Filter)
-	_, err := dockerClient.ListContainers()
-
-	if err != nil {
-		log.Fatalf("Could not connect to Docker Engine: %v", err)
+	for i := 1; ; i++ {
+		_, err := dockerClient.ListContainers()
+		if err == nil {
+			break
+		} else if !args.WaitForDocker {
+			log.Fatalf("Could not connect to Docker Engine: %v", err)
+		} else {
+			log.Infof("Waiting for Docker Engine (attempt %d): %s", i, err)
```

Comment:
Am I understanding this correctly that it can wait indefinitely? I don't think it should block forever. Common practice is to give up after some time or backoff using something like https://github.com/jpillora/backoff. 

What do you think?

Reply:
That works for me. I made a simple change to cap the total amount of waiting time in 9fc3936. Let me know if this works or if we want to pull in a third party dependency to have a fancier retry with more configuration options like polling interval, exponential back-off, etc. I didn't add all of these for now because I didn't want to pollute the settings with too many options until they actually become necessary (for my specific use-case, the simple timeout will be sufficient).

Reply:
I like this. I do prefer simple implementation over third party too.